### PR TITLE
OCD-3855: convert Announcements to have LocalDateTime

### PR DESCRIPTION
**Pull request by andlar**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/Announcement.java [hunk 1]
```diff
@@ -20,99 +42,21 @@
     private Long lastModifiedUser;
 
     public Announcement() {
+        super();
+    }
+
+    public Announcement(AnnouncementEntity entity) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/Announcement.java]: Do we still need this constructor AND the toDomain method on the entity? Todd prefers the former and I prefer the latter but we really only need 1 as far as I know.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/activity/GetActivityDetailsActionPermissions.java [hunk 2]
```diff
@@ -244,12 +244,12 @@ private boolean hasAccessToAnnualReport(final Long reportId, final JsonNode annu
...
+    private boolean hasAccessToAnnouncement(JsonNode announcementJson) {
         boolean hasAccess = false;
-        AnnouncementDTO announcement = null;
+        Announcement announcement = null;
         try {
-            announcement = jsonMapper.convertValue(announcementJson, AnnouncementDTO.class);
-        } catch (final Exception ex) {
+            announcement = jsonMapper.convertValue(announcementJson, Announcement.class);
+        } catch (Exception ex) {
             LOGGER.error("Could not parse announcement activity as AnnouncementDTO." + "JSON was: " + announcementJson,
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/activity/GetActivityDetailsActionPermissions.java]: Could you just change the log message to not say "AnnouncementDTO" as well? I know it's a small thing

**Review by kekey1 — DISMISSED** [review]
Thanks for removing a DTO!

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/AnnouncementManager.java [hunk 3]
```diff
@@ -2,33 +2,109 @@
...
 
-    AnnouncementDTO create(AnnouncementDTO announcement)
-            throws UserRetrievalException, EntityCreationException, EntityRetrievalException, JsonProcessingException;
+    @Autowired
+    private AnnouncementDAO announcementDAO;
 
-    AnnouncementDTO update(AnnouncementDTO announcement)
-            throws EntityRetrievalException, JsonProcessingException, EntityCreationException;
+    @Autowired
+    private ActivityManager activityManager;
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/AnnouncementManager.java]: Before we go to STG - these @Autowired variable should be moved to a constructor.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/activity/GetActivityDetailsActionPermissions.java [hunk 4]
```diff
@@ -31,17 +31,17 @@
 
 @Component("actionGetActivityDetailsActionPermissions")
 public class GetActivityDetailsActionPermissions extends ActionPermissions {
-    private static final Logger LOGGER = LogManager.getLogger(GetActivityDetailsActionPermissions.class);
+    private static Logger LOGGER = LogManager.getLogger(GetActivityDetailsActionPermissions.class);
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/activity/GetActivityDetailsActionPermissions.java]: Could this be moved to an `@Log4j2` annotation at the class level?

**Review by tmy1313 — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by kekey1 — CHANGES_REQUESTED** [review]
conflicts

**Review by tmy1313 — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**